syntax = "proto2";

// The Contacts service definition.
service Contacts {
  // Create a Contact
  rpc AddContact (Person) returns (TextMessage) {}

  // List Contacts
  rpc ListContacts (TextMessage) returns (AddressBook) {}

  // Find a Contact
  rpc FindContact (TextMessage) returns (Person) {}
}

message Person {
  required string name = 1;
  required int32 id = 2;
  optional string email = 3;

  enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }

  message PhoneNumber {
    required string number = 1;
    optional PhoneType type = 2 [default = HOME];
  }

  repeated PhoneNumber phone = 4;
}

message AddressBook {
  repeated Person person = 1;
}

message TextMessage {
    required string text = 1;
}